core/controllers/cron_test.py

```diff
@@ -316,6 +323,414 @@ def _mock_logging_function(msg, *args):
+            with self.swap(
+                suggestion_models,
+                'SUGGESTION_REVIEW_WAIT_TIME_THRESHOLD_IN_DAYS', 0):
+                with self.swap(
+                    email_manager,
+                    'send_mail_to_notify_admins_suggestions_waiting_long',
+                    self._mock_send_mail_to_notify_admins_suggestions_waiting):
+                    self.get_html_response(
+                        '/cron/mail/admins/contributor_dashboard_review_issues')
+
+        self.assertEqual(len(self.admin_ids), 0)
+        self.assertDictEqual(self.suggestion_types_need_reviewers, {})
+
+        self.logout()
+
+    def test_email_sent_to_admin_if_sending_admin_need_reviewers_emails_enabled(
+            self):
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+        config_services.set_property(
+            'committer_id', 'notify_admins_reviewers_needed_is_enabled', True)
+
+        with self.can_send_emails, self.testapp_swap:
+            with self.swap(
+                email_manager,
+                'send_mail_to_notify_admins_reviewers_needed',
+                self._mock_send_mail_to_notify_admins_reviewers_needed):
+                self.get_html_response(
+                    '/cron/mail/admins/contributor_dashboard_review_issues')
+
+        self.assertEqual(len(self.admin_ids), 1)
+        self.assertEqual(self.admin_ids[0], self.admin_id)
+        self.assertDictEqual(
+            self.suggestion_types_need_reviewers,
+            self.expected_suggestion_types_need_reviewers)
+
+    def test_email_sent_to_admin_if_notifying_admins_about_suggestions_enabled(
+            self):
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+        config_services.set_property(
+            'committer_id',
+            'notify_admins_suggestions_waiting_too_long_is_enabled', True)
+
+        with self.can_send_emails, self.testapp_swap:
+            with self.swap(
+                suggestion_models,
+                'SUGGESTION_REVIEW_WAIT_TIME_THRESHOLD_IN_DAYS', 0):
+                with self.swap(
+                    email_manager,
+                    'send_mail_to_notify_admins_suggestions_waiting_long',
+                    self._mock_send_mail_to_notify_admins_suggestions_waiting):
+                    self.get_html_response(
+                        '/cron/mail/admins/contributor_dashboard_review_issues')
+
+        self.assertEqual(len(self.admin_ids), 1)
+        self.assertEqual(self.admin_ids[0], self.admin_id)
+        self.assertEqual(
+            len(self.reviewable_suggestion_email_infos),
+            len(self.expected_reviewable_suggestion_email_infos))
+        for index, reviewable_suggestion_email_info in enumerate(
+                self.reviewable_suggestion_email_infos):
+            self._assert_reviewable_suggestion_email_infos_are_equal(
+                reviewable_suggestion_email_info,
+                self.expected_reviewable_suggestion_email_infos[index]
+            )
```

Comment:
If this isn't too long, prefer just listing out all the assertion lines rather than using a loop (prefer to avoid control flow in tests).

Reply:
changed